### PR TITLE
Reorder docs

### PR DESCRIPTION
**Pull request by ltratt**
This PR fleshes out the docs a bit more: in particular it adds new "parsing" and "lexing" sections, as well as putting the libraries and tools under an overall "libraries and tools" section.

There's still a lot more that could be done here, but this is probably the minimum I'm happy with.

### CODE REVIEW

File: doc/src/cfgrammar.md [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 `cfgrammar` reads in grammar files, processes them, and provides a convenient
 API for operating with them. It may be of interest to those manipulating
-grammars directly.
+grammars directly, or who to wish custom types of parsers.
```
Comment (ptersilie) [inline doc/src/cfgrammar.md]: `or wish to use custom types of parsers`?

Comment (ltratt) [reply doc/src/cfgrammar.md]: Oops!

**Comment by ltratt** [comment]
Pushed a fix for the boo-boo.

**Comment by ptersilie** [comment]
Looks good. If you squash I will merge and then rebase it in #55 as well.

**Comment by ltratt** [comment]
I've just pushed a commit adding me to the author list, so that you don't have to take the blame for any bits of nonsense that are my fault ;) Shall I squash the PR down after this?

**Comment by ptersilie** [comment]
Yes please! :-)

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**